net-core/Ical.Net/Ical.Net/Serialization/iCalendar/Serializers/Other/StringSerializer.cs

```diff
@@ -102,7 +102,7 @@ public override string SerializeToString(object obj)
             return string.Join(",", values);
         }
 
-        internal static readonly Regex UnescapedCommas = new Regex(@"[^\\](,)", RegexOptions.Compiled);
+        internal static readonly Regex UnescapedCommas = new Regex(@"(?<!\\),", RegexOptions.Compiled);
```

Comment:
It matches commas that are not preceded by backslashes.  The `(?<foo)` construct is a zero-width negative lookbehind assertion, which means it ensures the characters *before* that point don't match the given regex.  

Here: https://regex101.com/r/baxiSh/1